node/src/blockchain/blockchain_bridge.rs

```diff
@@ -963,46 +1202,61 @@ mod tests {
         let _ = addr.try_send(msg).unwrap();
 
         let system = System::new("transaction receipts");
-        System::current().stop();
         system.run();
         let accountant_recording = accountant_recording_arc.lock().unwrap();
-        assert_eq!(accountant_recording.len(), 1);
-        let received_message = accountant_recording.get_record::<ReportTransactionReceipts>(0);
+        assert_eq!(accountant_recording.len(), 2);
+        let report_transaction_receipt_message =
+            accountant_recording.get_record::<ReportTransactionReceipts>(0);
+        let scan_error_message = accountant_recording.get_record::<ScanError>(1);
+        let mut expected_receipt = TransactionReceipt::default();
+        expected_receipt.transaction_hash = hash_1;
         assert_eq!(
-            received_message,
+            report_transaction_receipt_message,
             &ReportTransactionReceipts {
                 fingerprints_with_receipts: vec![
                     (
-                        Some(TransactionReceipt::default()),
+                        TransactionReceiptResult::Found(expected_receipt),
                         pending_payable_fingerprint_1
                     ),
-                    (None, pending_payable_fingerprint_2),
+                    (
+                        TransactionReceiptResult::NotPresent,
+                        pending_payable_fingerprint_2
+                    ),
                 ],
                 response_skeleton_opt: Some(ResponseSkeleton {
                     client_id: 1234,
                     context_id: 4321
                 }),
             }
         );
-        let get_transaction_receipt_params = get_transaction_receipt_params_arc.lock().unwrap();
-        assert_eq!(*get_transaction_receipt_params, vec![hash_1, hash_2])
+        assert_eq!(
+            scan_error_message,
+            &ScanError {
+                scan_type: ScanType::PendingPayables,
+                response_skeleton_opt: Some(ResponseSkeleton {
+                    client_id: 1234,
+                    context_id: 4321
+                }),
+                msg: "Aborting scanning; 1 transactions succeed and 1 transactions failed"
+                    .to_string(),
+            }
+        )
     }
 
     #[test]
     fn blockchain_bridge_logs_error_from_retrieving_received_payments() {
         init_test_logging();
+        let port = find_free_port();
+        let _blockchain_client_server = MBCSBuilder::new(port)
+            .response("0x3B9ACA00".to_string(), 0)
```

Comment:
Add comment explaining that we have intentionally left out responses to cause this error